ingestors/rust-common/src/main.rs

```diff
@@ -135,6 +140,24 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         },
     };
 
+    let encryption_key = match args.encryption_key {
+        Some(e) => Ok(hmac::Key::new(hmac::HMAC_SHA256, e.as_bytes())),
+        None => match env::var("ENCRYPTION_KEY") {
+            Ok(s) => Ok(hmac::Key::new(hmac::HMAC_SHA256, s.as_bytes())),
+            Err(_) => {
+                let rng = ring::rand::SystemRandom::new();
+                hmac::Key::generate(hmac::HMAC_SHA256, &rng)
```

Comment:
Curious if this was required ? In case that a provided encryption key is not available, would it make sense to not send the encrypted part (as in just the trace metadata, not the trace itself). That way there won't be any unnecessary data transfer either.